apps/artemis-web/pages/api/rmq_action.ts

```diff
@@ -0,0 +1,150 @@
+      pass: RABBITMQ_PASS,
+      host: '172.18.0.16',
+      port: RABBITMQ_PORT,
+      timeout: 2000,
+      name: 'rabbitmq',
+    },
+    exchanges: [],
+    queues: [],
+    binding: [],
+    logging: {
+      adapters: {
+        stdOut: {
+          level: 3,
+          bailIfDebug: true,
+        },
+      },
+    },
+  };
+
+  let broker = new Broker(config);
+  await broker.connect();
+
+  await broker.addExchange(exchangeName, 'direct', {
+    durable: false,
+  } as BrokerExchangeOptions);
+  await broker.init();
+
+  console.log('sender: going to send message ..');
+  await broker.send(exchangeName, routing_key, payload.toString());
+  console.log('sender: message sent ..');
+};
+
+const handler = nc()
+  .use(auth)
+  .use(authorization(['admin']))
+  .post(async (req: NextApiRequestExtended, res: NextApiResponseExtended) => {
+    let obj = {};
+    const {
+      action,
+      hijack_key,
+      prefix,
+      hijack_type,
+      hijack_as,
+      state,
+    } = req.body;
+
+    switch (action) {
+      case 'mitigate':
+        obj = {
+          action: action,
+          routing_key: 'mitigate',
+          exchangeName: 'mitigation',
+          priority: 2,
+          payload: { key: hijack_key, prefix: prefix },
+        };
+        break;
+      case 'unmitigate':
+        obj = {
+          action: action,
+          routing_key: 'unmitigate',
+          exchangeName: 'mitigation',
+          priority: 2,
+          payload: { key: hijack_key, prefix: prefix },
+        };
+        break;
+      case 'resolve':
+        obj = {
+          action: action,
+          routing_key: 'resolve',
+          exchangeName: 'hijack-update',
+          priority: 2,
+          payload: {
+            key: hijack_key,
+            prefix: prefix,
+            type: hijack_type,
+            hijack_as: parseInt(hijack_as),
+          },
+        };
+        break;
+      case 'ignore':
+        obj = {
+          action: action,
+          routing_key: 'ignore',
+          exchangeName: 'hijack-update',
+          priority: 2,
+          payload: {
+            key: hijack_key,
+            prefix: prefix,
+            type: hijack_type,
+            hijack_as: parseInt(hijack_as),
+          },
+        };
+        break;
+      case 'delete':
+        obj = {
+          action: action,
+          routing_key: 'delete',
+          exchangeName: 'hijack-update',
+          priority: 2,
+          payload: {
+            key: hijack_key,
+            prefix: prefix,
+            type: hijack_type,
+            hijack_as: parseInt(hijack_as),
+          },
+        };
+        break;
+      case 'seen':
+        obj = {
+          action: action,
+          routing_key: 'seen',
+          exchangeName: 'hijack-update',
+          priority: 2,
+          payload: { key: hijack_key, state: state },
+        };
+        break;
+      default:
+        res.status(401);
+        return;
+    }
+
+    sendRMQAction(obj);
```

Comment:
the implementation seems fine code-wise, let's discuss this in person to find out what is going wrong.